### ISSUE TITLE
Favour "External IP" of a svc when running in external mode

### ISSUE THREAD

**Issue by sysadmiral** [haproxytech/kubernetes-ingress#356]
Hi,

Really appreciate the kubernetes-ingress! ❤️ 

I spoke with @Mo3m3n about this request on the slack channel.

When the controller is running outside the cluster `--external` and you have something in the cluster exposing IP addresses for you i.e. something like metallb then the controller should favour the allocated external ip instead of the svc's ip (which is in the pod subnet). This would also stop the need for `ip route add <pod-network> via <node-ip>` on the controller server since the external ip will likely be available on the local network.

So basically...

IF `--external` 
  AND svc `status.loadBalancer.ingress[*].ip` is not null
  THEN use `status.loadBalancer.ingress[*].ip` values in haproxy.cfg for backend location.

Hopefully that makes sense?

Cheers,

Amo

**Comment by nbraun-wolf** [comment]
Using the services ClusterIP would also be useful. For example to integrate with linkerd https://linkerd.io/2.10/tasks/using-ingress/#proxy-modes

**Comment by Mo3m3n** [comment]
Hi @sysadmiral 
Sorry for the delay, this is in the roadmap of the next release: 1.8 scheduled in two months

**Comment by sysadmiral** [comment]
I see we're up to `1.8.3` and wondered if this had been included? (it doesn't look like it yet from the release notes)

**Comment by sysadmiral** [comment]
Hi this has been open for 12 months. Is there any plan to look at adding this functionality/feature?

If it's unlikely then I can at least look at other ingresses.
